packages/app/src/cli/services/versions-list.ts

```diff
@@ -0,0 +1,123 @@
+import {fetchOrCreateOrganizationApp} from './context.js'
+import {AppDeploymentsQuery, AppDeploymentsQuerySchema} from '../api/graphql/get_versions_list.js'
+import {AppInterface} from '../models/app/app.js'
+import {getAppIdentifiers} from '../models/app/identifiers.js'
+import {partnersRequest} from '@shopify/cli-kit/node/api/partners'
+import {ensureAuthenticatedPartners} from '@shopify/cli-kit/node/session'
+import {renderTable} from '@shopify/cli-kit/node/ui'
+import colors from '@shopify/cli-kit/node/colors'
+import {outputContent, outputInfo, outputToken, unstyled} from '@shopify/cli-kit/node/output'
+
+// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
+type DeploymentLine = {
+  createdAt: string
+  createdBy?: string
+  message?: string
+  versionTag: string
+  status: string
+}
+
+function formatDate(date: Date) {
+  const components = date.toISOString().split('T')
+  const dateString = components[0] ?? date.toDateString()
+  const timeString = components[1]?.split('.')[0] ?? date.toTimeString()
+  return `${dateString} ${timeString}`
+}
+
+async function fetchDeployments(
+  token: string,
+  apiKey: string,
+): Promise<{deployments: DeploymentLine[]; totalResults: number; organizationId: string; appId: string}> {
+  const query = AppDeploymentsQuery
+  const res: AppDeploymentsQuerySchema = await partnersRequest(query, token, {apiKey})
+  const deployments = res.app.deployments.nodes.map((deployment) => {
```

Comment:
If the `appKey` does not exists (for example you deployed the app to spin) the API does not return any error but the `app` is undefined

![image](https://github.com/Shopify/cli/assets/105213827/9a5cd369-bd6d-49be-b1a0-d561b34b9383)